### PR TITLE
Add flang

### PR DESCRIPTION
**Pull request by xoviat**
I realize the the recipe is currently in a less-than-ideal condition. However, it will be much easier to work with once we get it into a feedstock and we're also working on resolving the upstream (clangdev) dependencies.

### CODE REVIEW

**Comment by xoviat** [comment]
@conda-forge-admin Please rerender.

**Comment by xoviat** [comment]
@isuruf Any idea why the appveyor build failed?

File: recipes/clangdev/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{% set clang_variant = os.environ.get('CLANG_VARIANT', 'flang') %}
+{% set flang_commit = "fd7f1c015c08ccaf66f6f2695aedf5d4a7961c07" %}
+{% set version = "4.0.0" %}
+{% set sha256 = "d9500ff12c050e51ba66b7ee336dc8279b80720cddd2f98ec3d27a22287a6683" %}
```
Comment (isuruf) [inline recipes/clangdev/meta.yaml]: You need to update this

**Comment by xoviat** [comment]
We still have a problem:

>  LINK : fatal error LNK1104: cannot open file 'flang.lib'

There is another library called "libflang.lib" created earlier.

**Comment by isuruf** [comment]
That's because of linker directives. We need to disable that when compiling flang

**Comment by xoviat** [comment]
@isuruf Is it okay if you merge this?

**Comment by isuruf** [comment]
No, until we get `clangdev` flang variant into conda-forge, it's not

**Comment by xoviat** [comment]
Okay, I guess we're stuck then. Sad.

**Pull request closed without merging**